mathics/builtin/numeric.py

```diff
@@ -824,7 +825,437 @@ def apply(self, n, base, evaluation, nr_elements=None):
+        else:
+            display_len = int(Expression('N', Expression('Round', Expression('Divide', Expression('Precision', py_n), Expression('Log', 10, py_b)))).evaluate(evaluation).to_python())
+        
+        exp = int(mpmath.ceil(mpmath.log(py_n, py_b))) if py_n != 0 and py_n != 1 else int(1)
+        
+        if py_n == 0 and nr_elements is not None:
+            exp = 0
+            
+        digits = [] 
+        if not py_b == 10: 
+            digits = convert_float_base(py_n, py_b, display_len - exp)
+            # truncate all the leading 0's
+            for i in range(0, len(digits)):
+                if(digits[0] != 0):
+                    break
+                else:
+                    digits = digits[1:]
+                    
+            if not isinstance(n, Integer):
+                if len(digits) > display_len:
+                    digits = digits[:display_len-1]    
+        else:
+            # drop any leading zeroes
+            for x in str(py_n):
+                if x != '.' and (digits or x != '0'):
+                    digits.append(x)
+                    
+        if pos is not None:
+            temp = exp
+            exp = pos + 1
+            move = temp - 1 - pos
+            if move <= 0:
+                digits = [0] * abs(move) + digits
+            else:
+                digits = digits[abs(move):]
+                display_len = display_len - move 
+                
+        list_str = Expression('List')
+        
+        for x in digits:
+            if x == 'e' or x == 'E':
+                break 
+            # Convert to Mathics' list format
+            list_str.leaves.append(from_python(int(x)))  
+        
+        if rational_no is not True:
+            while len(list_str.leaves) < display_len:
+                list_str.leaves.append(from_python(0))
+        
+        if nr_elements is not None:
+            # display_len == nr_elements
+            if len(list_str.leaves) >= nr_elements:
+                # Truncate, preserving the digits on the right
+                list_str = list_str.leaves[:nr_elements]   
+            else:
+                if isinstance(n, Integer):
+                    while(len(list_str.leaves) < nr_elements):
+                        list_str.leaves.append(from_python(0))
+                else:
+                    # Adding Indeterminate if the lenght is greater than the precision
+                    while(len(list_str.leaves) < nr_elements):
+                        list_str.leaves.append(from_python(Symbol('Indeterminate')))
+
+        return Expression('List', list_str, exp)
+    
+    def apply_3(self, n, b, length, evaluation, pos=None):
+        'RealDigits[n_?NumericQ, b_Integer, length_]'
+        
+        expr = Expression('RealDigits', n, b, length)
+        
+        if pos is not None: 
+            expr.leaves.append(from_python(pos))
+            
+        if not(isinstance(length, Integer) and length.get_int_value() >= 0):
```

Comment:
in case length.get_int_value() is None, it will not pass the check for isinstance(length, Integer) so it still returns the desired output.